src/layout/barGrid.ts

```diff
@@ -464,26 +515,77 @@ export function layout(seriesType: string, ecModel: GlobalModel) {
 
         lastStackCoords[stackId] = lastStackCoords[stackId] || [];
 
+        layoutInfo.groupOrder = seriesModel.get('groupOrder');
+        columnOffsetList[seriesIndex] = columnOffset;
+
         data.setLayout({
             bandWidth: columnLayoutInfo.bandWidth,
             offset: columnOffset,
             size: columnWidth
         });
-
         const valueDim = data.mapDimension(valueAxis.dim);
         const baseDim = data.mapDimension(baseAxis.dim);
         const stacked = isDimensionStacked(data, valueDim /*, baseDim*/);
         const isValueAxisH = valueAxis.isHorizontal();
 
         const valueAxisStart = getValueAxisStart(baseAxis, valueAxis, stacked);
-
+        let dataToPointList: number[];
         for (let idx = 0, len = data.count(); idx < len; idx++) {
```

Comment:
with typescript, is it now fair to comment the following syntax is preferred over C style for-loop?

 `for(let idx in data){`

https://www.typescriptlang.org/docs/handbook/iterators-and-generators.html#forof-vs-forin-statements

Reply:
OK, thanks for your advice,I will submit it again later to fix the problem.

Reply:
Can be used‘ zrUtil.each ’Instead of 'for (let idx in data) {'？

Reply:
'data' is not an array, it just uses the length obtained by `data.count ()` for data collection, and `for(let idx in data){` cannot be used